10.0 add choice model on layers

lmarion-source opened the pull request:
[IMP] Add the possibility to use different models on one geoengine map

lmarion-source committed:
[IMP] Add the possibility to superpose 2 layers from 2 different models on one map in geoengine

lmignon commented:
@lmarion-source Can you take the 10.0 branch as base and only add your last commit to create your PR. It seems that your branch is not bases on the last version of 10.0

lmignon commented on base_geoengine/static/src/js/views/geoengine_view.js:
```diff
@@ -229,39 +232,62 @@ var GeoengineView = View.extend(geoengine_common.GeoengineMixin, {
+        if (!!cfg.model_id){
+
+            var model = new Model(cfg.model_id[2]);
+            var fields_to_read = [cfg.geo_field_id[1]];
```
@lmarion-source In odoo, the field display_name is present by default on all the models. You should add this field by defaults into the list of retrieved fields and use this field as title of the details form dialog..

```js
var fields_to_read = [cfg.geo_field_id[1], 'display_name'];
```

lmignon commented on base_geoengine/static/src/js/views/geoengine_view.js:
```diff
@@ -788,18 +814,37 @@ var GeoengineView = View.extend(geoengine_common.GeoengineMixin, {
+                    res_model: model,
+                    res_id: oid,
+                    context: self.dataset.context,
+                    title: _t("Open: ") + attributes.name,
```
@lmarion-source the `name` attribute is not present by default. As proposed above, you should use `display_name`.

lmignon reviewed (changes requested):
@lmarion-source Greet feature!  some little comments...

lmarion-source committed:
add display name to fields + use it for box titles

lmignon commented:
@lmarion-source The name of the fields into the popup are not right when displaying the info of the layer defined for an other model. Into the fields definition given to the `formatFeatureHTML` is the one of the main model not the one of the layre's model.

lmignon committed:
[FIX] base_geoengine: Return the model name into the layer definition

lmignon committed:
[IMP] base_geoengine: Keep the layer config into the ol.Feature attributes

lmignon committed:
[IMP] base_geoengine: Allows to define a model on the raster layer to apply to the specified model

lmignon committed:
[IMP] base_geoengine: Allows to use template to display information for layers linked to an other model than the main one

lmignon committed:
[FIX] base_geoengine: Register event listener on popu element once the element is added to the overlay layer.
Before this change when the popup was dispalyed in front of an other geometry layer, events was handled by the geometry layer in place of the popu itself. To be sure that the events are first handled by the popup, the handlers must be registered once the dom element is added to the overlay layer

lmignon committed:
[IMP] base_geoengine: Style layers only once the data are loaded.

lmignon committed:
[FIX] avoid JS error if the map is rendered first on an invisible node and the value change.

To be properly initialized, the map must only be rendered on a visible node.

lmignon committed:
[FIX] base_geoengine: Only loads map when config is available
